### DISCUSSION TITLE
Overlaying SNOW extracted network and an image in ParaView

### DISCUSSION THREAD

**Discussion by Saeid-Sli** [PMEAL/porespy#514] [Q&A]
I want to overlay the SNOW extracted network and the original image in Paraview. I use the following code:

```python
prj.export_data(filename='extracted_network', filetype='vtk')
im = ps.tools.align_image_with_openpnm(im).astype(int)
ps.io.to_vtk(im, path='extracted_network')
```

However, the generated images are neither in the same size (the image is so larger than the network) nor in the same orientation. Would you please advise?

<img src="https://user-images.githubusercontent.com/29663268/115717133-2c546c80-a37a-11eb-84bf-3602a9bf9caa.png" width=50%>
<img src="https://user-images.githubusercontent.com/29663268/115717281-4ee68580-a37a-11eb-99aa-a05c6e363eb2.png" width=50%>

**Comment by jgostick** [comment]
Hi @Saeid-Sli 
You have to do two things.  Firstly, you the image needs to be rotated.  There is a function in porespy.tools called ``align_image_with_openpnm`` which does this for you.  Secondly, you need to tell paraview what are the voxel sizes of the tiff file.  There are 3 boxes in the pipeline editor for entering this, I forget what they're labelled but they're near the top.

**Reply by jgostick** [reply]
Ahh, I see that you already did the align with openpnm thing.  What versions are you using?

**Comment by Saeid-Sli** [comment]
Hi @jgostick 

I am using Paraview (5.7), the current version of PoreSpy, and OpenPNM (2.6.1)
I could find the resolution option in Paraview. Its name is "scale". However, after setting the scale, both images are even not lay on top of each other. See:
<img src="https://user-images.githubusercontent.com/29663268/115860687-228e4000-a432-11eb-9b92-16fe6e5726ca.png" width=50%>

However, after the rotation (90 degree around y-axis), I am not sure if the SNOW extraction was performed correctly. The visual comparison does not show that the location of pores are correct. There are some big pores (red ones) in the area of solid phases. plz check:

<img src="https://user-images.githubusercontent.com/29663268/115861639-4dc55f00-a433-11eb-984b-7016dc557079.png" width=50%>

Any ideas?

**Reply by ma-sadeghi** [reply]
Hi @Saeid-Sli, that's very strange. I tried and I didn't even need to rotate the image for them to align. I made a tutorial video a while back:

https://user-images.githubusercontent.com/14086031/115872761-2c06b080-a410-11eb-9f0b-910baf7652fa.mp4

Please let me know if this helps.

**Comment by Saeid-Sli** [comment]
Hi @ma-sadeghi 

Thank you for your help. I used the code in the movie. It seams that everything is working well.
The point was NOT using of the following line:
`im = ps.tools.align_image_with_openpnm(im).astype(int)`

![image](https://user-images.githubusercontent.com/29663268/115955590-bb3ac380-a4f7-11eb-90af-f34f5f1fe95f.png)


Just one more question. I use the default values of SNOW algorithm for running. But, I recieve the following message during the SNOW running:
"Maximum number of iterations reached, consider running again with a larger value of max_iters"

![image](https://user-images.githubusercontent.com/29663268/115955632-0f45a800-a4f8-11eb-9906-4ef40d5bc47a.png)

Is it a critical error? Because after the PNM extraction, I calculate its permeability by OpenPNM (5230 md) which is far from the permeability calculated by direct numerical simulation method on the same CT-image (~50-60 md).


Thanks

**Reply by ma-sadeghi** [reply]
I don't think that's a critical error. However, the mismatch is alarming! Can you share the CT image? (if you can't publicly share, maybe email)

**Reply by jgostick** [reply]
Some of the pores in that image look pretty small and I'm not fully confident that the snow algorithm would capture them all.  Could you try increasing the size of you image by 2x using scipy.ndimage.zoom, and re-run or would this make it too large?

**Reply by Saeid-Sli** [reply]
Hi @jgostick . 
I had a mistake in unit conversion. However, the updated K is now  K_openPNM >> K_DPS (5230 > 50-60md). Also, I have emailed both the structure and code to you and @ma-sadeghi.

**Reply by Saeid-Sli** [reply]
> 

@jgostick 
I used sp.ndimage.zoom with 2X zoom, but received more warning lines:
![image](https://user-images.githubusercontent.com/29663268/115968519-a29ecd80-a538-11eb-842d-0686adb44c2f.png)

**Reply by jgostick** [reply]
I forgot to mention that after zooming you have to convert image to a float, then back to a boolean.  Try to use a high order then threshold, like:
``im = spim.zoom(im.astype(float), zoom=3, factor=2) >= 1``

**Reply by Saeid-Sli** [reply]
> I forgot to mention that after zooming you have to convert image to a float, then back to a boolean. Try to use a high order then threshold, like:
> `im = spim.zoom(im.astype(float), zoom=3, factor=2) >= 1`


Do you mean?

`sp.ndimage.zoom(im.astype(float), zoom=3, order=2) >= 1`

I used the above code, and during the distance transform calculation, a new odd error crashes the spyder (I am using a system with German windows :). "Programm schleißen" means "Close the program"). 

![image](https://user-images.githubusercontent.com/29663268/115971105-edbfdd00-a546-11eb-83c8-77b19ddbee2e.png)
